mlrun/projects/project.py

```diff
@@ -3546,6 +3547,77 @@ def remove_custom_packager(self, packager: str):
         """
         self.spec.remove_custom_packager(packager=packager)
 
+    def create_or_update_api_gateway(
+        self, api_gateway: mlrun.runtimes.nuclio.api_gateway.APIGateway
+    ) -> mlrun.runtimes.nuclio.api_gateway.APIGateway:
+        """
+        Creates or updates a Nuclio API Gateway using the provided APIGateway object.
+
+        This method interacts with the MLRun service to create/update a Nuclio API Gateway based on the provided
+        APIGateway object. Once done, it returns the updated APIGateway object containing all fields propagated
+        on MLRun and Nuclio sides, such as the 'host' attribute.
+        Nuclio docs here: https://docs.nuclio.io/en/latest/reference/api-gateway/http.html
+
+        :param api_gateway: An instance of mlrun.runtimes.nuclio.APIGateway representing the configuration
+        of the API Gateway to be created
+
+        @return: An instance of mlrun.runtimes.nuclio.APIGateway with all fields populated based on the
+        information retrieved from the Nuclio API
+        """
+
+        api_gateway_json = mlrun.db.get_run_db().store_api_gateway(
+            api_gateway=api_gateway,
+            project=self.name,
+        )
+
+        if api_gateway_json:
+            # fill in all the fields in the user's api_gateway object
+            api_gateway = mlrun.runtimes.nuclio.api_gateway.APIGateway.from_scheme(
+                api_gateway_json
+            )
+        return api_gateway
+
+    def list_api_gateways(self) -> list[mlrun.runtimes.nuclio.api_gateway.APIGateway]:
+        """
+        Retrieves a list of Nuclio API gateways associated with the project.
+
+        @return: List of mlrun.runtimes.nuclio.api_gateway.APIGateway objects representing the Nuclio API gateways
+        associated with the project.
+        """
+        gateways_list = mlrun.db.get_run_db().list_api_gateways(self.name)
+        return [
+            mlrun.runtimes.nuclio.api_gateway.APIGateway.from_scheme(gateway_dict)
+            for gateway_dict in gateways_list.api_gateways.values()
+        ]
+
+    def get_api_gateway(
+        self,
+        name: Optional[str],
+        api_gateway: Optional[mlrun.runtimes.nuclio.api_gateway.APIGateway],
+    ) -> mlrun.runtimes.nuclio.api_gateway.APIGateway:
```

Comment:
@TomerShor it was my idea with a future extension. It might be useful to request actual `Status` of gateway. Also, user might update the api-gateway in one jupyter instance and upload the changes in another instance.

Reply:
Okay, then will leave only the name.